app/views/editorial/nodes/show.html.haml

```diff
@@ -47,6 +47,11 @@
     %tr
       %td State
       %td= node.state
+    -# TODO: refactor this to properly handle different options per Node subclass
+    - if node.options.respond_to?(:toc)
```

Comment:
Is node here decorated? If so, perhaps you could define a new method on the decorator like:

    def has_toc?
      object.options.respond_to?(:toc)
    end